### ISSUE TITLE
Updater.exe Blocked by Windows Defender

### ISSUE THREAD

**Issue by hl2guide** [Rem0o/FanControl.Releases#2133]
**Describe the bug**
Updater.exe Blocked by Windows Defender

![ex](https://github.com/Rem0o/FanControl.Releases/assets/17044982/68ee258b-f56a-48f4-b068-f961e11d1f2e)

**Comment by bczegeny** [comment]
Same but it thought it was another trojen
![image](https://github.com/Rem0o/FanControl.Releases/assets/13109866/38ba3e19-ddf9-4538-bfb5-560082ec912d)

**Comment by ungkal96** [comment]
got this message too
![Screenshot (7)](https://github.com/Rem0o/FanControl.Releases/assets/138473208/4d39e533-4c59-480f-bd44-eead94f8d131)

**Comment by CIsxxc** [comment]
I get the exact same as the above....

**Comment by jilherme** [comment]
same

**Comment by agarrandosenal** [comment]
same Wacatac.B!ml

Sad to have to find an alternative as I don't think this is a false positive. End of an era.

**Comment by ErisaFirehawk** [comment]
Trojan:Script/Wacatac.B!ml

well damn

**Comment by CIsxxc** [comment]
Wait, he does mention false positives here, but can @Rem0o comment on this?

![image](https://github.com/Rem0o/FanControl.Releases/assets/92247173/a5ca6fdd-8314-49ed-b1bd-322cd91b6a5d)

**Comment by Rem0o** [comment]
Whenever I recompile the updater, it triggers Microsoft Defender.   I send a false positive submission, it gets scanned, then the false positive stops.

![image](https://github.com/Rem0o/FanControl.Releases/assets/14118956/28e472d1-fc8b-4234-8eaa-9830a578febe)

**Comment by ad0x00** [comment]
Same here, got the threat quarantined. @Rem0o should I restore the updater or delete and reinstall FC after some time?

**Comment by CIsxxc** [comment]
> Whenever I recompile the updater, it triggers Microsoft Defender.   I send a false positive submission, it gets scanned, then the false positive stops.
> 
> 
> 
> ![image](https://github.com/Rem0o/FanControl.Releases/assets/14118956/28e472d1-fc8b-4234-8eaa-9830a578febe)
> 
> 

Thank you so much for your time and the effort you put into this project.

I'll be donating to you this payday after having used the app for a couple months very happily!

Have a lovely day/evening :)

**Comment by Spuner** [comment]
Windows Defender is a misunderstanding.

**Comment by a-zndr** [comment]
Importantly if you're here because your FanControl isn't running now, and hangs on launch. the update process hasn't fully completed. redownload fancontrol and reinstall all the files as a replacement, preserving your config and plugins folder.

unfortunately because they are binaries you cannot confirm there isn't actually a trojan here. @Rem0o could be a compromised account encouraging us to download.

**Comment by marrok657** [comment]
> Importantly if you're here because your FanControl isn't running now, and hangs on launch. the update process hasn't fully completed. redownload fancontrol and reinstall all the files as a replacement, preserving your config and plugins folder.
> 
> unfortunately because they are binaries you cannot confirm there isn't actually a trojan here. @Rem0o could be a compromised account encouraging us to download.

I have tried to download the newest and it still tells me "threat detected". unless maybe I download the version before?

**Comment by tullahstackz** [comment]
I had the same issue as everyone else. I finally got it working after I ran "Updater.exe". The first time it was run is shown in the attached screenshot. After deleting the .dll file that is mentioned, I ran Updater.exe again and the program runs as usual.
![Screenshot (368)](https://github.com/Rem0o/FanControl.Releases/assets/61850773/7b21002e-48cc-4826-83a9-9d5a241b1bc3)

**Comment by csandazoltan** [comment]
Until Windows Defender Sorts out its detection of the new updater, save your configurations, download 174 directly from the site.

**Comment by Trae132** [comment]
> Until Windows Defender Sorts out its detection of the new updater, save your configurations, download 174 directly from the site.

Thats what worked for me although I didnt try the poster above's recommendation of deleting the file, mainly because I didnt see the update.exe file referred to in their post.  I copied the json and plugin folder from old install into new downloaded one from the website and all good.

**Comment by AussieGomez** [comment]
Yup same with me tried to update and just install straight from website and defender grabs it as well.

**Comment by Khanivore** [comment]
I had the same problem but now fant control won't start at all.

**Comment by yvesfouquet4** [comment]
> I had the same issue as everyone else. I finally got it working after I ran "Updater.exe". The first time it was run is shown in the attached screenshot. After deleting the .dll file that is mentioned, I ran Updater.exe again and the program runs as usual. ![Screenshot (368)](https://user-images.githubusercontent.com/61850773/284786678-7b21002e-48cc-4826-83a9-9d5a241b1bc3.png)

this solved it, thanks

**Comment by danielbr93** [comment]
It quarantined it for me, but after downloading the new update, unzipping it, deleting everything but the "Config" folder in the old folder and copy pasting the update into it, everything seems to work again.
If the Trojan is still doing stuff in the background, I can't tell.

**Comment by supabibz** [comment]
> I had the same issue as everyone else. I finally got it working after I ran "Updater.exe". The first time it was run is shown in the attached screenshot. After deleting the .dll file that is mentioned, I ran Updater.exe again and the program runs as usual. ![Screenshot (368)](https://user-images.githubusercontent.com/61850773/284786678-7b21002e-48cc-4826-83a9-9d5a241b1bc3.png)

Worked for me too : had to delete Autofac.dll & Newtonxxxx.dll. After running the update, those dll didn't reappear...

**Comment by Chicora470** [comment]
There`s some news about this update is false positive or an hacked false update that bring virus?

**Comment by Chicora470** [comment]
> Importantly if you're here because your FanControl isn't running now, and hangs on launch. the update process hasn't fully completed. redownload fancontrol and reinstall all the files as a replacement, preserving your config and plugins folder.
> 
> unfortunately because they are binaries you cannot confirm there isn't actually a trojan here. @Rem0o could be a compromised account encouraging us to download.

No response from @Rem0o about this?

**Comment by AtA3301** [comment]
Not had any issues with Windows Defender here.  Remi already addressed why this can happen above: https://github.com/Rem0o/FanControl.Releases/issues/2133#issuecomment-1821942006

Defender (and most AV products) will flag a program that tries to modify itself (ie, update) if the signature is not known. When FC is re-compiled, the signature is changed, and there will always be a delay between release and Defender being up to date.

**Comment by travislrogers** [comment]
Just encountered the same issue although it didn't mention 'trojan', but 'HackTool'. Will try re-insalling it from a fresh download.
